.azure-pipelines/ci-build.yml

```diff
@@ -318,119 +325,127 @@ extends:
                   assets: '$(Pipeline.Workspace)\**\*.exe'
                   addChangeLog: false
 
-    - stage: Build_and_deploy_docker_images
-      displayName: 'Build and deploy docker images'
-      condition: or(eq(variables['build.sourceBranch'], 'refs/tags/v'), eq(variables['build.sourceBranch'], variables['PREVIEW_BRANCH']))
-      dependsOn: build
-      pool:
-        name: Azure-Pipelines-1ESPT-ExDShared
-        image: ubuntu-latest
-        os: linux
-      jobs:
-        - job: buildAndPush
-          steps:
-            - task: AzureCLI@2
-              displayName: 'Login to Azure Container Registry'
-              inputs:
-                azureSubscription: 'ACR Images Push Service Connection'
-                scriptType: bash
-                scriptLocation: inlineScript
-                inlineScript: |
-                  az acr login --name msgraphprodregistry
-            
-            - powershell: |
-                $content = [XML](Get-Content ./Directory.Build.props)
-                Write-Host "XML loaded, finding version..."
-                
-                # Handle PropertyGroup as either a single element or array
-                $version = $null
-                if ($content.Project.PropertyGroup -is [array]) {
-                  Write-Host "PropertyGroup is an array, checking each entry..."
-                  foreach ($pg in $content.Project.PropertyGroup) {
-                    if ($pg.Version) {
-                      $version = $pg.Version.ToString().Trim()
-                      Write-Host "Found version in PropertyGroup array: $version"
-                      break
+      - deployment: deploy_docker_image
+        environment: kiota-github-releases
+        strategy:
+          runOnce:
+            deploy:
+              pool:
+                vmImage: 'ubuntu-latest'
+              steps:
+                - checkout: self
```

Comment:
we might not be allowed to checkout the repository in a deployment stage, have you tested this?

Reply:
It actually works. Let me remove the checkout step

Reply:
Also, why is this not allowed?

Reply:
deployment phases are not allowed to use checkout steps per security policy. Effectively they don't want scripts that could lead to some kind of injection I suppose.
If we revert to a regular job (not a deployment one) it'd be allowed though.
I have no doubt that it can work, but I'm fairly confident it'll trigger security items down the road.

Reply:
Got it! Thanks